packages/openapi-v3/src/extension-point/spec-contributor.service.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/example-greeter-extension
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {config, Getter} from '@loopback/context';
+import {extensionPoint, extensions} from '@loopback/core';
+import * as debugModule from 'debug';
+import {inspect} from 'util';
+import {OpenApiSpec} from '../types';
+import {
+  OAISpecContributor,
+  OAISPEC_CONTRIBUTOR_EXTENSION_POINT_NAME,
+} from './types';
+const debug = debugModule('loopback:openapi:spec-contributor');
+
+/**
+ * Options for the OAI Spec contributor extension point
+ */
+export interface OAISpecContributorServiceOptions {
+  // no-op
+}
+
+/**
+ * An extension point for OAI contributors
+ */
+@extensionPoint(OAISPEC_CONTRIBUTOR_EXTENSION_POINT_NAME)
+export class SpecService {
+  constructor(
+    /**
+     * Inject a getter function to fetch spec contributors
+     */
+    @extensions()
+    private getContributors: Getter<OAISpecContributor[]>,
+    /**
+     * An extension point should be able to receive its options via dependency
+     * injection.
+     */
+    @config()
+    public readonly options?: OAISpecContributorServiceOptions,
+  ) {}
+
+  private _spec: OpenApiSpec = {
+    openapi: '3.0.0',
+    info: {
+      title: 'LoopBack Application',
+      version: '1.0.0',
+    },
+    paths: {},
+  };
+
+  /**
+   * Find contributors for a given field
+   * @param fieldName - The field name
+   */
+  async findContributors(): Promise<OAISpecContributor[] | undefined> {
```

Comment:
This method can be removed to use `this.getContributors()` directly.

Reply:
right. created it for filtering by field, not needed any more.